src/test/java/net/snowflake/client/jdbc/ResultSetAsyncIT.java

```diff
@@ -308,4 +308,25 @@ public void testIsSerializable() throws SQLException {
       // Do nothing. Test passes if we catch this exception.
     }
   }
+
+  @Test
+  public void testEmptyResultSet() throws SQLException {
+    Connection connection = getConnection();
+    Statement statement = connection.createStatement();
+    ResultSet rs =
+        statement.unwrap(SnowflakeStatement.class).executeAsyncQuery("select * from empty_table");
+    // if user never calls getMetadata() or next(), empty result set is used to get results.
+    // empty ResultSet returns all nulls, 0s, and empty values.
+    assertFalse(rs.isClosed());
+    assertEquals(0, rs.getInt(1));
+    try {
+      assertEquals(0, rs.getInt("col1"));
+      fail("Fetching from a column name that does not exist should return a SQLException");
```

Comment:
Could you please show me why this `rs.getInt("col1")` will throw an exception?

Reply:
It is an empty ResultSet object. However, in this case, the ResultSetMetaData is also empty (because real results have not been fetched). The ResultSetMetaData object is where the mapping of column names is contained. When getInt(String columnName) is called, it attempts to match a column name to a column index using the ResultSetMetaData's mapping. This mapping is empty, so the findColumn() function returns an exception saying the column name cannot be found.

In the other test case I added here with getGeneratedKeys() where we don't encounter this exception, it's because getGeneratedKeys() returns SnowflakeResultSetV1.getEmptyResultSet() directly with no attempt to fetch any real results, so it's just returning the hard-coded values from that class. This approach of returning just an empty ResultSet would not work with SFAsyncResultSet because then we'd be unable to fetch the real data when it exists.

Reply:
I see, it calls getInt() in SnowflakeBaseResultSet. That makes sense. Thanks a lot for the explanation!